upload_artifacts.sh

```diff
@@ -1,21 +1,33 @@
 #!/bin/bash
 
-# Used in .travis.yml
+# This script is used in .travis.yml
+
+set -e
 
 BUCKET=piksi-buildroot-images
 PIKSI_VERSION=v3
+BUILD_VERSION=$(git rev-parse --short HEAD)
 
-set -e
+if [ "$TRAVIS_PULL_REQUEST" == "true" ]; then
+  FOLDER=pull_requests
+else
+  if [ "$TRAVIS_BRANCH" == "master" ]; then
+    FOLDER=master
+  else
+    FOLDER=misc
+  fi
+fi
```

Comment:
this logic seems a little complex - how about something simpler:

```bash
FOLDER="$TRAVIS_BRANCH"
if [ "$TRAVIS_PULL_REQUEST" == "true" ]; then
FOLDER="pr/$FOLDER"
fi
```

That way there's a convention where the branch is always the folder, and there's an optional pr folder up front.